### ISSUE TITLE
Not receiving notification

### ISSUE THREAD

**Issue by sachins-geekyants** [atsign-foundation/at_client_sdk#1256]
### Describe the bug

Not receiving notification while running atPro.
I tried multiple platforms(ios, macos, windows) with multiple atSigns.

**In last deployed version of atPro, notifications are coming.**

at_client versions
```
at_client_mobile : "3.2.15"
at_client :  "3.0.75"
```

### Steps to reproduce

1. Run the app
2. send file from another app
3. notifications would not come up
4. on refresh the data shows up in history screen

### Expected behavior

Notifications should be coming

### Additional context

Attaching logs

```
flutter: INFO|2024-03-06 19:47:30.292447|SyncService (@mightysanddollar)|541227884|Returning serverCommitId 10392
flutter: INFO|2024-03-06 19:47:30.563195|SyncService (@mightysanddollar)|541227884|Returning serverCommitId 10392
flutter: INFO|2024-03-06 19:47:30.942328|SyncService (@mightysanddollar)|541227884|Inside syncComplete. syncRequest.requestSource : SyncRequestSource.system; syncRequest.onDone : Closure: (SyncResult) => void from Function '_onDone@3242025363':.
flutter: syncStatus type : Sync status: SyncStatus.success lastSyncedOn: 2024-03-06 14:17:30.942309Z Exception: null, datachanged : true
flutter: INFO|2024-03-06 19:47:31.255470|SyncService (@mightysanddollar)|541227884|Returning serverCommitId 10392
flutter: WARNING|2024-03-06 19:47:40.520899|Monitor (@mightysanddollar)|runZonedGuarded received socket error Converting object to an encodable object failed: Instance of 'Metadata' - calling _handleError
flutter: INFO|2024-03-06 19:47:40.522920|OutboundConnectionImpl|close(): calling socket.destroy() on connection to InternetAddress('35.239.254.23', IPv4):12590
flutter: INFO|2024-03-06 19:47:40.523526|Monitor (@mightysanddollar)|socket.listen onDone called. Will destroy socket, set status stopped, call retryCallback
flutter: INFO|2024-03-06 19:47:40.525644|Monitor (@mightysanddollar)|Monitor error Converting object to an encodable object failed: Instance of 'Metadata' - calling the retryCallback
flutter: INFO|2024-03-06 19:47:40.525871|NotificationServiceImpl (@mightysanddollar)|Monitor retry already queued
flutter: INFO|2024-03-06 19:47:46.744974|Monitor (@mightysanddollar)|monitor started for @mightysanddollar with last notification time: null
flutter: INFO|2024-03-06 19:48:00.284937|SyncService (@mightysanddollar)|541227884|Returning serverCommitId 10393
flutter: INFO|2024-03-06 19:48:00.587172|SyncService (@mightysanddollar)|541227884|Returning serverCommitId 10393
flutter: INFO|2024-03-06 19:48:00.976598|SyncService (@mightysanddollar)|541227884|Inside syncComplete. syncRequest.requestSource : SyncRequestSource.system; syncRequest.onDone : Closure: (SyncResult) => void from Function '_onDone@3242025363':.
flutter: syncStatus type : Sync status: SyncStatus.success lastSyncedOn: 2024-03-06 14:18:00.976575Z Exception: null, datachanged : true
flutter: INFO|2024-03-06 19:48:01.243238|SyncService (@mightysanddollar)|541227884|Returning serverCommitId 10393
flutter: WARNING|2024-03-06 19:55:14.339516|Monitor (@mightysanddollar)|runZonedGuarded received socket error Converting object to an encodable object failed: Instance of 'Metadata' - calling _handleError
flutter: INFO|2024-03-06 19:55:14.340480|OutboundConnectionImpl|close(): calling socket.destroy() on connection to InternetAddress('35.239.254.23', IPv4):12590
flutter: INFO|2024-03-06 19:55:14.341435|Monitor (@mightysanddollar)|socket.listen onDone called. Will destroy socket, set status stopped, call retryCallback
flutter: INFO|2024-03-06 19:55:14.341949|Monitor (@mightysanddollar)|Monitor error Converting object to an encodable object failed: Instance of 'Metadata' - calling the retryCallback
flutter: INFO|2024-03-06 19:55:14.342203|NotificationServiceImpl (@mightysanddollar)|Monitor retry already queued
flutter: INFO|2024-03-06 19:55:21.781175|Monitor (@mightysanddollar)|monitor started for @mightysanddollar with last notification time: null
```

**Comment by sitaram-kalluri** [comment]
```
flutter: WARNING|2024-03-06 19:47:40.520899|Monitor (@mightysanddollar)|runZonedGuarded received socket error Converting object to an encodable object failed: Instance of 'Metadata' - calling _handleError
flutter: INFO|2024-03-06 19:47:40.522920|OutboundConnectionImpl|close(): calling socket.destroy() on connection to InternetAddress('35.239.254.23', IPv4):12590
flutter: INFO|2024-03-06 19:47:40.523526|Monitor (@mightysanddollar)|socket.listen onDone called. Will destroy socket, set status stopped, call retryCallback
flutter: INFO|2024-03-06 19:47:40.525644|Monitor (@mightysanddollar)|Monitor error Converting object to an encodable object failed: Instance of 'Metadata' - calling the retryCallback
```

Snippet from the above logs. The error occurs because of a bug in at_commons - 4.0.2. The bug is fixed in at_commons 4.0.3.

@sachins-geekyants : Can you please upgrade at_commons to 4.0.3 version and try.

**Comment by sachins-geekyants** [comment]
Thanks @sitaram-kalluri
